spinner/runner/instance_runner.py

```diff
@@ -20,238 +25,129 @@ def __str__(self):
+        retry: int | None = None,
+    ) -> None:
+        """Run a command and capture its output."""
+        stdout, stderr, retcode, elapsed = self.launch_process_with_retry(
+            command, timeout, retry
         )
 
-        return parameter_combinations
-
-    def get_run_instruction(self) -> list:
-        instructions = []
-
-        parameter_combinations = self.get_parameter_combinations()
-
-        for parameters in list(parameter_combinations):
-            rprint(f"self.metadata: {self.metadata}")
-            command_info = self.metadata[self.bench_name]["command"]
-            template_string = command_info["template"]
-            rprint(f"command_template: {template_string}")
-
-            env = Environment(undefined=StrictUndefined)
-            template = env.from_string(template_string)
-
-            command = template.render(
-                self.metadata | parameters | self.metadata[self.bench_name]
-            )
-            rprint(f"rendered command: {command}")
-
-            instructions.append(
-                {
-                    "cmd": command,
-                    "parameters": parameters,
-                }
-            )
-
-        return instructions
-
-    def check_output(self, output) -> bool:
-        # TODO check the output
-        rprint("[red] Check output not implemented!")
-        return True
-
-    def filter_output(self, output, error, output_filter):
-        combined_output = str(output + error)
-        filtered_output = ""
-        captured_parameters = {}
-
-        for filter in output_filter:
-            current_capture = ""
-
-            if filter["type"] == "all":
-                current_capture = combined_output
-
-            elif filter["type"] == "contains":
-                rprint(f"-> Filtering lines that contain: {filter['pattern']}")
-                current_capture += "\n".join(
-                    [
-                        line
-                        for line in combined_output.split("\n")
-                        if filter["pattern"] in line
-                    ]
-                )
-                current_capture += "\n"
-
-            else:
-                rprint(f"[red] Unsupported filter type: {filter['type']}")
-
-            if "to_float" in filter:
-                rprint(f"-> Converting captured line to float")
-                param_name = filter["to_float"]["name"]
-                param_lambda = filter["to_float"]["lambda"]
-
-                try:
-                    param_value = eval(param_lambda, {}, {})(current_capture)
-                except Exception as e:
-                    rprint(f"-> Error evaluating param_lambda: {e}")
-                    param_value = None
-
-                captured_parameters[param_name] = param_value
+        if retcode != 0:
+            self.app.error("Failed to run command.")
+            return
 
-            filtered_output += current_capture
+        captures = self.process_captures(stdout)
```

Comment:
What do you think about merging application output?
```suggestion
        captures = self.process_captures('\n'.join([stdout, stderr]))
```

Reply:
Great idea! Will change in the next commit.